src/app/dataset-view/dataset.component.ts

```diff
@@ -411,62 +427,47 @@ export class DatasetComponent
-        }
+        this.getDatasetInfoFromUrl();
 
-        if (searchParams.length > 1) {
-            const type: DatasetViewTypeEnum = AppValues.fixedEncodeURIComponent(
-                searchParams[1].split("&")[0],
-            ) as DatasetViewTypeEnum;
+        this.appDatasetService.resetDatasetTree();
 
-            this.datasetViewType = type;
-            if (type === DatasetViewTypeEnum.overview) {
-                this.initOverviewTab();
-            }
-            if (type === DatasetViewTypeEnum.data) {
-                this.initDataTab();
-            }
-            if (type === DatasetViewTypeEnum.metadata) {
-                this.initMetadataTab(page);
-            }
-            if (type === DatasetViewTypeEnum.history) {
-                this.initHistoryTab(page);
-            }
-            if (type === DatasetViewTypeEnum.linage) {
-                this.initLineageTab();
-            }
-            if (type === DatasetViewTypeEnum.discussions) {
-                this.initDiscussionsTab();
-            }
+        if (this.datasetViewType === DatasetViewTypeEnum.overview) {
+            this.initOverviewTab();
+        }
+        if (this.datasetViewType === DatasetViewTypeEnum.data) {
+            this.initDataTab();
+        }
+        if (this.datasetViewType === DatasetViewTypeEnum.metadata) {
+            this.initMetadataTab(this.currentPage);
+        }
+        if (this.datasetViewType === DatasetViewTypeEnum.history) {
+            this.initHistoryTab(this.currentPage);
+        }
+        if (this.datasetViewType === DatasetViewTypeEnum.linage) {
+            this.initLineageTab();
+        }
+        if (this.datasetViewType === DatasetViewTypeEnum.discussions) {
+            this.initDiscussionsTab();
         }
     }
 
-    private getDatasetId(): string {
-        const searchParams: string[] = decodeURIComponent(
-            this.searchString,
-        ).split("?id=");
-
-        if (searchParams.length > 1) {
-            return searchParams[1].split("&")[0];
-        }
-        return "";
+    private getDatasetInfoFromUrl(): void {
+        this.datasetInfo.accountName =
+            this.activatedRoute.snapshot.paramMap.get("accountName") || "";
+        this.datasetInfo.datasetName =
+            this.activatedRoute.snapshot.paramMap.get("datasetName") || "";
     }
 
     public onSelectDataset(id: string): void {
-        this.navigationService.navigateToDatasetView(
-            AppValues.defaultUsername,
-            id,
-            ProjectLinks.urlDatasetViewLineageType,
-        );
+        if (this.datasetBasics) {
+            this.navigationService.navigateToDatasetView({
+                accountName: AppValues.defaultUsername,
+                datasetName: this.selectedDatasetName
+                    ? this.selectedDatasetName
+                    : this.datasetBasics?.name,
+                tab: ProjectLinks.urlDatasetViewLineageType,
+            });
+        }
```

Comment:
This function looks weird now.
1) It is not using the given ID
2) Above I've mentioned, that instead of ID, it should be getting a name from the node
3) Maybe `accountName` can also be transported instead of  `defaultUsername`

Reply:
Modified method